### PR TITLE
DOC-648 | Stable multi-dimensional indexes

### PR DESCRIPTION
**Pull request by Simran-B**
### Description

- [x] Feature list
- [x] Release notes
- [x] Indexes
- [x] HTTP API

#### Upstream PRs

- 3.10:
- 3.11:
- 3.12: arangodb/enterprise-preview:devel-nightly

### CODE REVIEW

File: site/content/3.12/develop/http-api/indexes/multi-dimensional.md [hunk 1]
```diff
@@ -48,24 +48,61 @@ paths:
...
+                    `storedValues` is not considered by index creation calls when checking if an
+                    index is already present or needs to be created.
+
+                    In unique indexes, only the attributes in `fields` are checked for uniqueness,
+                    but the attributes in `storedValues` are not checked for their uniqueness.
+                    Non-existing attributes are stored as `null` values inside `storedValues`.
+
+                    Attributes in `storedValues` cannot overlap with attributes
+                    specified in `prefixFields` but you can have the attributes
+                    in both `storedValues` and `fields`.
```
Comment (maierlars) [inline site/content/3.12/develop/http-api/indexes/multi-dimensional.md:86]: We could a word about why you want to have attributes in `fields` **and** `storedValues`: you can not project from `fields`. A typical use case would be to have `fields: ["x", "y"]` and `storedValues: ["x", "y", "foo"]` if you want to work with the `x` and `y` attributes of the document. If they are not in stored values, we have to load the whole document, which is a huge performance penalty.

**Review by maierlars — COMMENTED** [review]
Maybe add more details about how `prefixFields` work? At first their use might be a bit counter intuitive.

File: site/content/3.12/index-and-search/indexing/working-with-indexes/multi-dimensional-indexes.md [hunk 2]
```diff
@@ -7,15 +7,47 @@ description: >-
...
+
+Multi-dimensional indexes can be created with a uniqueness constraint with
+`unique: true`. This only allows a single document with a given combination of
+attribute values, using all of the `fields` attributes and (for `mdi-prefixed`
+indexes) `prefixFields`. Documents omitted because of `sparse: true` are exempt.
+
+You can store additional attributes in multi-dimensional indexes with the
+`storedValues` property. They can be used for projections (unlike the `fields`
+attributes) so that indexes can cover more queries without having to access the
+full documents.
```
Comment (maierlars) [inline site/content/3.12/index-and-search/indexing/working-with-indexes/multi-dimensional-indexes.md:37]: Just as a side node, while the persistent index does not allow having an overlap between `storedValues` and `fields`, the mdi index explicitly allows this. However, `prefixFields` and `storedValues` should not overlap. A typical pattern is to have `fields = ['x', 'y']` and `storedValues= ['x', 'y']` if you want to project from `x` and `y` later on.

File: site/content/3.12/index-and-search/indexing/working-with-indexes/multi-dimensional-indexes.md [hunk 3]
```diff
@@ -123,11 +155,103 @@ Thus our query would be:
...
+## Prefix fields
+
+Multi-dimensional indexes can accelerate range queries well but they are
+inefficient for queries that check for equality of values. For use cases where
+you have a combination of equality and range conditions in queries, you can use
+the `mdi-prefixed` subtype instead of `mdi`. It has all the features of the
+`mdi` subtype but additionally lets you define one or more document attributes
+you want to use for equality checks. This combination allows to efficiently
+narrow down the search space to a subset of multi-dimensional index data before
+performing the range checking.
```
Comment (maierlars) [inline site/content/3.12/index-and-search/indexing/working-with-indexes/multi-dimensional-indexes.md:172]: The attribute values in the `prefixFields` can also be non-numeric. Which usually was a huge problem with the old `mdi`.

File: site/content/3.12/index-and-search/indexing/working-with-indexes/multi-dimensional-indexes.md (hunk 2 above)
Comment (Simran-B) [reply site/content/3.12/index-and-search/indexing/working-with-indexes/multi-dimensional-indexes.md:37]: Added a link to the section

**Review by nerpaula — APPROVED** [review]

**Pull request merged**